sdk/nodejs/automation/stack.ts

```diff
@@ -510,14 +510,21 @@ Event: ${line}\n${e.toString()}`);
 
         let logPromise: Promise<ReadlineResult> | undefined;
         let logFile: string | undefined;
+        let loggedSummary: OpMap | undefined;
+
         // Set up event log tailing
-        if (opts?.onEvent) {
-            const onEvent = opts.onEvent;
+        if (opts?.previewOnly || opts?.onEvent) {
             logFile = createLogFile("refresh");
             args.push("--event-log", logFile);
 
             logPromise = this.readLines(logFile, (event) => {
-                onEvent(event);
+                if (opts?.previewOnly && event.summaryEvent) {
+                    loggedSummary = event.summaryEvent.resourceChanges;
```

Comment:
What else is in `summaryEvent`? Does the rest of it make sense in these cases? E.g. we could just do `summary = event.summaryEvent`, potentially, and save the `info` call entirely (I feel like `preview` does this?) -- but maybe we want a hybrid of info from the preview and the last operation?

Reply:
Actually, this is a bit of a pain. We need the following for an update summary:

```yaml
- Kind # we can hardcode this per operation
- Message # empty string is fine
- StartTime # event has a timestamp?
- ResourceChanges # from event summary
- Environment # ?
- Config # ?
- Result # Can just be "succeeded"
- EndTime # startTime + event summary duration?
- Version # ?
```

You're right - doing what I did definitely doesn't actually work, because all the non-resourceChanges info is still wrong. Maybe that's fine for some of these points, but definitely not for all of them. 🤔